packages/core/src/Site/index.js

```diff
@@ -1466,6 +1496,147 @@ class Site {
+    this.siteConfig.ignore.push(path.join(pathToVersionFromRootDir, VERSIONS_DATA_NAME));
+
+    const pathsToVersionFiles
+     = walkSync(pathToDirWithVersion, { directories: false, ignore: this.siteConfig.ignore })
+       .filter(x => x.endsWith(VERSIONS_DATA_NAME))
+       .map(x => path.relative(pathToDirWithVersion, x)) // assumes versions files are in the 'root' of site
+       .map(x => path.dirname(x));
+
+    pathsToVersionFiles.forEach(p => this.ignoreVersionFiles(p));
+  }
+
+  /**
+   * Reads version data from the version file.
+   *
+   * @param {string} versionDataFile, default is VERSIONS_DATA_NAME at the root
+   * @returns a json object
+   */
+  async readVersionData(versionDataFile = VERSIONS_DATA_NAME) {
+    const versionsPath = path.join(this.rootPath, versionDataFile);
+    try {
+      if (!fs.pathExistsSync(versionsPath)) {
+        // Initialize the versions.json file.
+        fs.outputJSONSync(versionsPath, { versions: [] }, { spaces: 2 });
+      }
+      return fs.readJSON(versionsPath);
+    } catch (error) {
+      await Site.rejectHandler(error, [this.tempPath, this.outputPath]);
+      return null;
+    }
+  }
+
+  /**
+   * If the versions.json file exists, update it with a new version.
+   * Otherwise, create a new versions file to store information about archived versions
+   *
+   * @param {boolean} verbose Flag to emit logs of the operation
+   * @returns Returns the json object of the current versions data.
+   */
+  async writeVersionsFile(versionName, archivePath, verbose = true) {
+    const newVersionData = {
+      versionName,
+      buildVer: MARKBIND_VERSION,
+      archivePath,
+      baseUrl: this.siteConfig.baseUrl,
+    };
+
+    try {
+      const versionsJson = await this.readVersionData(VERSIONS_DATA_NAME);
+
+      // Add in or update this new version data in the versions file.
+      const idx = versionsJson.versions.findIndex(vers => vers.versionName === newVersionData.versionName);
+      if (idx === -1) {
+        versionsJson.versions.push(newVersionData);
+      } else {
+        versionsJson.versions[idx] = newVersionData;
+      }
+      fs.writeJsonSync(VERSIONS_DATA_NAME, versionsJson, { spaces: 2 });
+      if (verbose) {
+        logger.info('versions.json file updated');
+      }
+
+      return versionsJson;
+    } catch (error) {
+      await Site.rejectHandler(error, [this.tempPath, this.outputPath]);
+      return null;
+    }
+  }
+
+  /**
+   * Copies over all versioned files from a given folder to be deployed.
+   * @param {Array} versionFolders is the directory the versions are within
+   */
+  async addVersions(versionFolders) {
+    const versionFoldersArray = versionFolders.map(f => f.archivePath);
+    try {
+      versionFoldersArray.map(async (versionFolder) => {
+        fs.copy(versionFolder, path.join(SITE_FOLDER_NAME, versionFolder));
+      });
+      await Promise.all(versionFoldersArray);
+      logger.info('Pages copied');
```

Comment:
```suggestion
      logger.info('Versioned sites copied');
```

So that it is clearer to the user on what is being copied.